### PR TITLE
refactor: move pkg/extensions/search/common/oci_layout.go un…

### PR DESCRIPTION
**Pull request by nicoldr**
…der pkg/test/

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/test/oci_layout.go [hunk 1]
```diff
@@ -24,14 +28,16 @@ type OciLayoutUtils interface { //nolint: interfacebloat
...
 	GetImagePlatform(imageInfo ispec.Image) (string, string)
 	GetImageManifestSize(repo string, manifestDigest godigest.Digest) int64
-	GetRepoLastUpdated(repo string) (TagInfo, error)
-	GetExpandedRepoInfo(name string) (RepoInfo, error)
+	GetRepoLastUpdated(repo string) (common.TagInfo, error)
+	GetExpandedRepoInfo(name string) (common.RepoInfo, error)
 	GetImageConfigInfo(repo string, manifestDigest godigest.Digest) (ispec.Image, error)
 	CheckManifestSignature(name string, digest godigest.Digest) bool
 	GetRepositories() ([]string, error)
+	ExtractImageDetails(ctx context.Context, repo, tag string, log log.Logger) (godigest.Digest,
```
Comment (andaaron) [inline pkg/test/oci_layout.go:39]: The context was removed from the implementation, but not here?

File: pkg/test/oci_layout.go [hunk 2]
```diff
@@ -550,3 +556,27 @@ func (olu BaseOciLayoutUtils) GetExpandedRepoInfo(repoName string) (RepoInfo, er
 
 	return repo, nil
 }
+
+func (olu BaseOciLayoutUtils) ExtractImageDetails(
+	repo, tag string, //nolint:unparam // function only called in the tests
```
Comment (andaaron) [inline pkg/test/oci_layout.go:561]: This nolint comment should not be needed.

Comment (nicoldr) [reply pkg/test/oci_layout.go]: Comment removed

**Review by andaaron — APPROVED** [review]

**Review by andaaron — DISMISSED** [review]

**Comment by nicoldr** [comment]
> @nicoldr why do we need pkg/search/common/ anymore now?

We have here the logic for tags and labels, also the functions to get all kinds of details contained in these labels and tags

**Comment by rchincha** [comment]
> > @nicoldr why do we need pkg/search/common/ anymore now?
> 
> We have here the logic for tags and labels, also the functions to get all kinds of details contained in these labels and tags

"common" with what? typically represents some shared code with other parts of the codebase. If this is confined to search/ alone, would just move it up to under pkg/search/ itself.

**Comment by nicoldr** [comment]
> > > @nicoldr why do we need pkg/search/common/ anymore now?
> > 
> > 
> > We have here the logic for tags and labels, also the functions to get all kinds of details contained in these labels and tags
> 
> "common" with what? typically represents some shared code with other parts of the codebase. If this is confined to search/ alone, would just move it up to under pkg/search/ itself.

Hello, Ram. I tried today the scenario in which we move all the code inside common directly under search, but import problems appeared. I had a meeting with Andrei and a new issue was created, and there you can see propositions to move for all the parts that are under search/common right now. Can you take a look at the issue and see if the suggested changes seem alright to you? New issue: #1342

Since it will be quite a work to do there, that PR would be independent from this one and can be reviewd separately

**Review by rchincha — APPROVED** [review]
lgtm

Ok let this be first part of the changes

**Pull request merged**